v0/destinations/pipedrive/transform.js

```diff
@@ -0,0 +1,416 @@
+  // name is required field. If name is not present, construct payload will
+  // throw error
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  payload = extractCustomFields(
+    message,
+    payload,
+    ["traits", "context.traits"],
+    PIPEDRIVE_IDENTIFY_EXCLUSION
+  );
+
+  payload = renameCustomFields(payload, destination.Config.fieldsMap, "person");
+
+  const userIdValue = getFieldValueOrThrowError(
+    message,
+    "userId",
+    new Error("userId is required")
+  );
+  const person = await searchPersonByCustomId(userIdValue, destination);
+
+  // update person since person already exists
+  if (person) {
+    const response = defaultRequestConfig();
+    response.method = defaultPutRequestConfig.requestMethod;
+    response.headers = {
+      "Content-Type": "application/json",
+      Accept: "application/json"
+    };
+
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    response.endpoint = `${PERSONS_ENDPOINT}/${person.id}`;
+    response.params = {
+      api_token: destination.Config.api_token
+    };
+
+    return response;
+  }
+
+  // mapping userId to custom userId key field
+  // in destination payload
+  set(payload, destination.Config.userIdKey, userIdValue);
+
+  // create a new person
+  const response = defaultRequestConfig();
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+  response.endpoint = PERSONS_ENDPOINT;
+  response.params = {
+    api_token: destination.Config.api_token
+  };
+
+  return response;
+};
+
+/**
+ * for group call, extracting from both traits and context.traits
+ * VERIFY ONCE
+ * @param {*} message
+ * @param {*} category
+ * @param {*} destination
+ * @returns
+ */
+const groupResponseBuilder = async (message, category, destination) => {
+  // name is required field. If name is not present, construct payload will
+  // throw error
+
+  const groupId = getFieldValueOrThrowError(
+    message,
+    "groupId",
+    new Error("groupId is required for group call")
+  );
+
+  let groupPayload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  groupPayload = removeUndefinedAndNullValues(groupPayload);
+
+  groupPayload = extractCustomFields(
+    message,
+    groupPayload,
+    ["traits", "context.traits"],
+    PIPEDRIVE_GROUP_EXCLUSION
+  );
+
+  groupPayload = renameCustomFields(groupPayload, destination.Config.fieldsMap, "organization");
+
+  let org = await searchOrganisationByCustomId(groupId, destination);
+
+  /**
+   * check if the person actually exists
+   * either userId or anonId is required for group call
+   */
+  const userIdVal = getFieldValueOrThrowError(
+    message,
+    "userId",
+    new Error("error: userId is required")
+  );
+
+  const person = await searchPersonByCustomId(userIdVal, destination);
```

Comment:
do this call before searching for organisation? if this only fails no need for finding org